### PR TITLE
360 - Refactor Dataset Page to handle dataset ID via URL query

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number

Epic: #358 

Closing #360 and #362

Feature branch: `feature/epic-refactor-dataset-page-data-handling`

## Purpose/Implementation Notes

I've made the following changes to the dataset page and its sub-components:
- Managed dataset data via URL query
- Removed states for various dataset types
- Moved the dataset status variables to sub-component level and simplify JSX
- Added check to redirect users if `dataset/id` matches their My Dataset ID

## Types of changes

- Refactor (addresses code organization and design mentioned in corresponding issue)

## Functional tests

N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

N/A

### CODE REVIEW

**Review by davidsmejia — COMMENTED** [review]
Wrote a couple of commnets. I'd like to improve how we distinguish between a dataset from a prop and the myDataset and when / if it matters. Let me know your thoughts on my suggestion. Otherwise the bulk of the PR looks correct to me.

File: src/components/Dataset/MoveToDatasetButton.js
[the remark this answers is not in the document]
Comment (nozomione) [reply src/components/Dataset/MoveToDatasetButton.js:1]: Thank you for these great suggestions 👍 I totally agree with keeping the prop as `dataset` and using `dataset.data`. Renaming the `dataset` context state to `myDataset` is an excellent idea for clarity and readability as well! Let’s discuss this further in our 1:1, and once we finalize the implementation strategy, we can file a new issue for it Looking forward!

**Comment by nozomione** [comment]
I've applied your feedback and this PR is ready for another look. Thank you David!

**Comment by nozomione** [comment]
I've updated the naming of the user-defined dataset to `myDataset`  at the component level. I'll file a new issue for the renaming of the `dataset` and `datasetId` context states (for user-saved data in `LocalStorage`) once we finalize the naming for `datasetId` etc. 

I've also filed the issue for the `getDatasetState` helper which came up during our 1:1, please see #370.

This PR is now ready for your review. Thank you David!

File: src/components/Dataset/DatasetPageHeader.js [hunk 1]
```diff
@@ -35,16 +34,18 @@ const Block = ({ children }) => {
...
   const { setResponsive } = useResponsive()
+  const {
+    expires_on: expiredOn,
+    is_available: isAvailable,
+    is_processed: isProcessed,
+    is_processing: isProcessing,
+    success
+  } = dataset
+  const isExpired = moment(expiredOn).isBefore(Date.now())
+  const isProcessingError = success === false // 'success' may be null
```
[the remark this answers is not in the document]
Comment (nozomione) [reply src/components/Dataset/DatasetPageHeader.js:45]: David, as chat on Slack (and noted in here as well https://github.com/AlexsLemonade/refinebio-web/pull/364#issuecomment-2377780601), I've filed a dedicated new issue (#370, for adding `getDatasetState` helper and refactoring the components that use dataset states. This should make the review easier while  keeping the file changes minimal and reducing the merge conflicts (the PR for it is also ready #373). Thank you David!

File: src/pages/dataset/[dataset_id].js [hunk 2]
```diff
@@ -82,52 +80,43 @@ export const Dataset = ({ query }) => {
...
-        <Box align="center" fill justify="center" margin={{ top: 'large' }}>
-          <Spinner />
-        </Box>
-      ) : (
-        <Box>
-          {selectedDataset?.data && (
-            <>
-              <DatasetPageHeader dataset={selectedDataset} />
+      <Box>
+        {dataset?.data && (
```
Comment (davidsmejia) [inline src/pages/dataset/[dataset_id].js:94]: I think we can get rid of this check as well there should only be 2 states we care about. Loading and loaded. I would think that if dataset.data is undefined then loading would be true.

**Review by davidsmejia — APPROVED** [review]
One comment, otherwise everything else I would suggest should be handled in a future PR.

File: src/pages/dataset/[dataset_id].js (hunk 2 above)
Comment (nozomione) [reply src/pages/dataset/[dataset_id].js]: It has been cleaned up in 41e82602de98496f22dfcb909804e15277db76f3 👍

**Pull request merged**